### PR TITLE
fix(go): shorten file names for sub-packages

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
CDKTF is running into build issues one some providers due to file names being too long for golang to handle: https://github.com/cdktf/cdktf-provider-googlebeta/actions/runs/4002506018/jobs/6869839656#step:8:11

This is my slightly naive try to get rid of a few characters (especially with long package names)

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

**Comment by DanielMSchmidt** [comment]
I have a hard time updating the snapshots 🙈 Can someone help me out here?

File: packages/jsii-pacmak/lib/targets/go/package.ts [hunk 1]
```diff
@@ -266,23 +263,9 @@ export abstract class Package {
...
 
-    emit.call(
-      this,
-      join(
-        this.directory,
-        `${this.packageName}_${type.name}__runtime_type_checks.go`,
-      ),
-      false,
-    );
+    emit.call(this, join(this.directory, `${type.name}__checks.go`), false);
```
Comment (RomainMuller) [inline packages/jsii-pacmak/lib/targets/go/package.ts:266]: Does this not risk collisions if two namespaces have the same leaf type name?

**Review by RomainMuller — COMMENTED** [review]
To update snapshots, you should just be able to run `yarn test:update`

File: packages/jsii-pacmak/lib/targets/go/package.ts (hunk 1 above)
Comment (DanielMSchmidt) [reply packages/jsii-pacmak/lib/targets/go/package.ts:266]: No, they are in nested folders

**Review by RomainMuller — DISMISSED** [review]

**Review by RomainMuller — APPROVED** [review]

**Pull request merged**